flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceSql.java

```diff
@@ -39,4 +62,76 @@ public void before() throws IOException {
         .getConfiguration()
         .set(TableConfigOptions.TABLE_DYNAMIC_TABLE_OPTIONS_ENABLED, true);
   }
+
+  private Record generateRecord(Instant t1) {
+    Record record = GenericRecord.create(SCHEMA_TS);
+    record.setField("t1", t1.atZone(ZoneId.systemDefault()).toLocalDateTime());
+    record.setField("t2", t1.getEpochSecond());
```

Comment:
What I do not like in this test, that if we make a mistake and switch `t1` with `t2` in somewhere in the production code, or in the test then we will not see a difference in the test results.

That is why I originally suggested, that the different columns have different values to make sure that the `watermark-column` value is propagated correctly

Reply:
To make sure that the watermark generation is set, the test checks that the ordering of the splits is correct. We create 2 files add them to the table. If the watermark generation is not set, then the records from the files are returned in the same order as we added to the table. If the watermark generation is set, then the splits are returned in the order based on the data.

If the data is the same in `t1` and in `t2` columns, then there is no difference when we order by (generate watermarks) based on `t1` or `t2`.  So the test is less restrictive.

Reply:
> If the watermark generation is not set, then the records from the files are returned in the same order as we added to the table. If the watermark generation is set, then the splits are returned in the order based on the data.

both test methods enable watermark generation. 

> If the data is the same in t1 and in t2 columns, then there is no difference when we order by (generate watermarks) based on t1 or t2. So the test is less restrictive.

I guess I probably understand why previously t1 and t2 are set to different values. you want the order to be different depnding on t1 or t2 is the watermark column.

Reply:
the unit test is using the 2 columns approach, so that we can cover the scenarios described by @pvary